### PR TITLE
[RFR] Verify that upon customer deletion, associated users and providers are also deleted

### PR DESCRIPTION
**Pull request by nachandr**
Results: 
All the tests passed locally.

### CODE REVIEW

File: hansei/tests/api/conftest.py [hunk 1]
```diff
@@ -3,38 +3,74 @@
...
+@pytest.fixture(scope='function')
 def service_admin(koku_config):
-    return KokuServiceAdmin(username = koku_config.get('username'), password = koku_config.get('password'))
+    return KokuServiceAdmin(username=koku_config.get('username'),
+                            password=koku_config.get('password'))
+
+
+class HanseiBaseTestAPI(object):
+    @pytest.fixture(scope='class')
+    def koku_config_classscope(self):
```
Comment (LandoCalrizzian) [inline hansei/tests/api/conftest.py]: You won't need to rename this fixture. It's local to the class so it takes naming precedence.

File: hansei/tests/api/conftest.py [hunk 2]
```diff
@@ -3,38 +3,74 @@
...
+        return config.get_config().get('koku', {})
+
+    @pytest.fixture(scope='class')
+    def service_admin_classscope(self, koku_config_classscope):
```
Comment (LandoCalrizzian) [inline hansei/tests/api/conftest.py]: You won't need to rename this fixture. It's local to the class so it takes naming precedence. 

Update the koku_config argument name

File: hansei/tests/api/v1/test_customer.py [hunk 3]
```diff
@@ -1,59 +1,49 @@
...
 
         for cust in customer_list:
-            assert cust.uuid != customer.uuid, "Customer was not deleted from the koku server"
+            assert cust.uuid != new_customer.uuid, \
+                "Customer was not deleted from the koku server"
+
+        # TODO : In future, verify that the data has been deleted from the DB
 
+        new_user.login()
+        current_user = new_user.get_current_user()
```
Comment (LandoCalrizzian) [inline hansei/tests/api/v1/test_customer.py]: You won't need to login() AND get_current_user(). Either one can verify that authentication will fail. 
Also you need to catch the exception that will be thrown when authentication fails

File: hansei/tests/api/conftest.py [hunk 4]
```diff
@@ -3,36 +3,61 @@
...
+from hansei.koku_models import KokuServiceAdmin
+
+
+class HanseiBaseTestAPI(object):
+    @pytest.fixture(scope='class')
+    def koku_config(self):
+        return config.get_config().get('koku', {})
+
+    @pytest.fixture(scope='class')
+    def service_admin(self, koku_config):
```
[the remark this answers is not in the document]
Comment (nachandr) [reply hansei/tests/api/conftest.py]: Added function scoped fixtures as well.

File: hansei/tests/api/v1/test_customer.py [hunk 5]
```diff
@@ -1,59 +1,52 @@
...
 
         for cust in customer_list:
-            assert cust.uuid != customer.uuid, "Customer was not deleted from the koku server"
+            assert cust.uuid != new_customer.uuid, \
+                "Customer was not deleted from the koku server"
+
+        # TODO : In future, verify that the user and provider data has been
+        # deleted from the DB
 
+        try:
```
Comment (LandoCalrizzian) [inline hansei/tests/api/v1/test_customer.py]: OPTIONAL. In the future you can use "with pytest.raises(HTTPError)"

**Pull request merged**